pkg/envoy/registry/services_test.go

```diff
@@ -63,13 +70,55 @@ var _ = Describe("Test Proxy-Service mapping", func() {
 
 			expectedSvc2 := service.MeshService{
 				Namespace: tests.Namespace,
-				Name:      svcName2,
+				Name:      fmt.Sprintf("%s.%s", podName, svcName2),
 				Port:      tests.ServicePort,
 				Protocol:  "http",
 			}
+
+			// Subdomain gets called in the ListProxyServices
+			expectedSvc1.Subdomain()
+			expectedSvc2.Subdomain()
 			expectedList := []service.MeshService{expectedSvc1, expectedSvc2}
 
-			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(nil, nil).Times(2)
+			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(&v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: tests.Namespace,
+					Name:      expectedSvc1.Name,
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{
+							{
+								IP: "8.8.8.8",
+							},
+							{
+								IP: "8.8.8.9",
+							},
+						},
+					},
+				},
+			}, nil).Times(1)
+
+			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(&v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: tests.Namespace,
+					Name:      expectedSvc2.Name,
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{
+							{
+								IP:       "8.8.8.9",
+								Hostname: podName,
+							},
+							{
+								IP:       "8.8.8.9",
+								Hostname: podName2,
```

Comment:
Is there any issue with not actually creating the pod with this name in the test here if it's not referred to otherwise?

Reply:
There's technically no pod being persisted in a cluster; it's just allocated in memory and returned as the result of a mock function (e.g. `ListPods`)

Reply:
I mean it looks like the test can sidestep even creating the mocked pod as long as the name is still defined here. That would remove some code from the test that otherwise isn't particularly relevant.

Reply:
Yes, but only for the proxy that's being passed to `ListProxyServices`. The second pod is only ever needed in the endpoints here. That goes back to the whole "realistic vs. minimal" thing though.

Reply:
Wait, correct me if I'm wrong...but `listPodsForService` isn't ever used (even on the main branch). If that's the case, you're right, there's no need for the full pod manifest. I'll remove the function and clean this up

Reply:
Well, turns out that's not where `ListPods` was being called, but I'll just defer to you on this one 😄